### ISSUE TITLE
Can you provide more details on intermediate steps in your example model(s) pipeline?

### ISSUE THREAD

**Issue by cudmore** [devnums/pyheartlib#13]
Can we get more details on the intermediate steps in your analysis pipeline?

For example, your web docs have a model page 
   https://pyheartlib.readthedocs.io/en/latest/examples/model/model.html

This page first links 'beat detection' and has headings : 
 - Data Preparation
 - Training
 - Inference
 - Results

It results in a [plot](https://github.com/devnums/pyheartlib/blob/main/examples/model/plots/mis.png).

Can you provide more detailed documentation about each of these intermediate steps? I would treat each of your headings/steps like a function. For example: what is the input, what is the computation, and what is the output?

I suggest this documentation could be a jupyter notebook with descriptive markdown, code blocks, and importantly **intermediate plots** of the data starting with the raw data.

Basically, what exactly is happening at and between  each of the steps including 'data preparation', 'training', 'inference', 'results', and especially the final plot? What is the input, computation, and output of each of these steps?

**Comment by devnums** [comment]
As per your suggestion, I have included additional information for each step. The example is now available as a Jupyter Notebook and can also be run in the Google Colab (Colab link is included at the top of the file). Please let me know if I have added enough information. 
https://pyheartlib.readthedocs.io/en/latest/examples/rpeak_detection.html
